docs/src/imagesegmentation.md

```diff
@@ -0,0 +1,490 @@
+# ImageSegmentation.jl
+
+## Introduction
+
+Image Segmentation is the process of partitioning the image into regions that have similar attributes. Image segmentation has various applications e.g, medical image segmentation, image compression and is used as a preprocessing step in higher level vision tasks like object detection and optical flow. This package is a collection of image segmentation algorithms written in Julia.
+
+## Installation
+
+```julia
+Pkg.add("ImageSegmentation")
+```
+
+## Example
+
+Image segmentation is not a mathematically well-defined problem: for example, the only lossless representation of the input image would be to say that each pixel is its own segment. Yet this does not correspond to our own intuitive notion that some pixels are naturally grouped together. As a consequence, many algorithms require parameters, often some kind of threshold expressing your willingness to tolerate a certain amount of variation among the pixels within a single segment.
+
+Let's see an example on how to use the segmentation algorithms in this package. We will try to separate the horse, the ground and the sky in the image below. We will explore two algorithms - seeded region growing and felzenszwalb. Seeded region growing requires us to know the number of segments and some points on each segment beforehand whereas felzenszwalb uses a more abstract parameter controlling degree of within-segment similarity.
+
+![Original](assets/segmentation/horse.jpg)
+
+[source](https://ibb.co/hUQMnQ)
+
+The documentation for `seeded_region_growing` says that it needs two arguments - the image to be segmented and a set of seed points for each region. The seed points have to be stored as a vector of `(position, label)` tuples, where `position` is a [`CartesianIndex`](https://docs.julialang.org/en/stable/manual/arrays/#Cartesian-indices-1) and `label` is an integer. We will start by opening the image using ImageView and reading the coordinates of the seed points.
+
+```julia
+using Images, ImageView
+
+img = load("horse.jpg")
+imshow(img)
+```
+
+Hover over the different objects you'd like to segment, and read out the coordinates of one or more points inside each object. We will store the seed points a vector of `(seed position, label)` tuples and use `seeded_region_growing` with the recorded seed points.
+
+```julia
+using ImageSegmentation
+seeds = [(CartesianIndex(126,81),1), (CartesianIndex(93,255),2), (CartesianIndex(213,97),3)]
+segments = seeded_region_growing(img, seeds)
+```
+
+All the segmentation algorithm (except Fuzzy C-means) return a struct `SegmentedImage` that stores the segmentation result. `SegmentedImage` contains a list of applied labels, an array containing the assigned label for each pixel and mean color and number of pixels in each segment. [This section](##Result) explains how to access information about the segments. 
```

Comment:
"All the segmentation algorithm*s* ..." (pluralize)

Also, add comma: "...for each pixel, and mean color and number of pixels..."

The Result link doesn't quite work. (You're presumably aware you can test via `include("make.jl")` in the top directory and then examining `build/index.html` in your browser.)